grafana-plugin/src/components/ScheduleQuality/ScheduleQuality.tsx

```diff
@@ -1,96 +1,131 @@
+import { Tooltip, VerticalGroup } from '@grafana/ui';
 import cn from 'classnames/bind';
 
+import PluginLink from 'components/PluginLink/PluginLink';
+import ScheduleCounter from 'components/ScheduleCounter/ScheduleCounter';
+import { ScheduleQualityDetails } from 'components/ScheduleQualityDetails/ScheduleQualityDetails';
+import Tag from 'components/Tag/Tag';
 import Text from 'components/Text/Text';
+import { Schedule, ScheduleScoreQualityResponse, ScheduleScoreQualityResult } from 'models/schedule/schedule.types';
+import { useStore } from 'state/useStore';
 
-import styles from './ScheduleQuality.module.css';
+import styles from './ScheduleQuality.module.scss';
+
+const cx = cn.bind(styles);
 
 interface ScheduleQualityProps {
-  quality: number;
+  schedule: Schedule;
+  lastUpdated: number;
 }
 
-const cx = cn.bind(styles);
+const ScheduleQuality: FC<ScheduleQualityProps> = ({ schedule, lastUpdated }) => {
+  const { scheduleStore } = useStore();
+  const [qualityResponse, setQualityResponse] = useState<ScheduleScoreQualityResponse>(undefined);
 
-const ScheduleQuality: FC<ScheduleQualityProps> = (props) => {
-  const { quality } = props;
+  useEffect(() => {
+    if (schedule.id) {
+      fetchScoreQuality();
+    }
+  }, [schedule.id, lastUpdated]);
 
-  return (
-    <Tooltip placement="bottom-end" interactive content={<SheduleQualityDetails quality={quality} />}>
-      <div className={cx('root')}>
-        <HorizontalGroup spacing="sm">
-          <Text type="secondary">Quality:</Text>
-          <Text type="primary">{Math.floor(quality * 100)}%</Text>
-        </HorizontalGroup>
-      </div>
-    </Tooltip>
-  );
-};
+  if (!qualityResponse) {
+    return null;
+  }
 
-interface ScheduleQualityDetailsProps {
-  quality: number;
-}
+  const relatedEscalationChains = scheduleStore.relatedEscalationChains[schedule.id];
 
-const SheduleQualityDetails = (props: ScheduleQualityDetailsProps) => {
-  const { quality } = props;
+  return (
+    <>
+      <div className={cx('root')}>
+        {relatedEscalationChains?.length > 0 && schedule?.number_of_escalation_chains > 0 && (
+          <ScheduleCounter
+            type="link"
+            addPadding
+            count={schedule.number_of_escalation_chains}
+            tooltipTitle="Used in escalations"
+            tooltipContent={
+              <VerticalGroup spacing="sm">
+                {relatedEscalationChains.map((escalationChain) => (
+                  <div key={escalationChain.pk}>
+                    <PluginLink query={{ page: 'escalations', id: escalationChain.pk }} className={cx('link')}>
+                      <Text type="link">{escalationChain.name}</Text>
+                    </PluginLink>
+                  </div>
+                ))}
+              </VerticalGroup>
+            }
+          />
+        )}
 
-  const [expanded, setExpanded] = useState<boolean>(false);
+        {schedule.warnings?.length > 0 && (
```

Comment:
can schedule be `null/undefined` here?

Reply:
No, if it's `undefined` then we do not render `ScheduleQuality`